test/cpp_extensions/open_registration_extension/torch_openreg/csrc/runtime/OpenRegDeviceAllocator.cpp

```diff
@@ -1,8 +1,275 @@
 #include "OpenRegDeviceAllocator.h"
+#include "OpenRegFunctions.h"
+
+#include <c10/util/Exception.h>
+#include <c10/util/irange.h>
+
+using namespace c10::CachingAllocator;
 
 namespace c10::openreg {
 
-static OpenRegDeviceAllocator global_openreg_alloc;
-REGISTER_ALLOCATOR(c10::DeviceType::PrivateUse1, &global_openreg_alloc);
+constexpr size_t kAggregate = static_cast<size_t>(StatType::AGGREGATE);
+
+
+DeviceMemoryAllocator::DeviceMemoryAllocator(c10::DeviceIndex device_index)
```

Comment:
I prefer to move the simple and concise implementation to the header file (this can be done in the next PR).